subscriptions/middleware.py

```diff
@@ -6,6 +9,14 @@
 
 class SubscriptionsMiddleware(MiddlewareMixin):
     def process_request(self, request):
-        is_auth = request.user.is_authenticated
+        if settings.HEADER_VALIDATION_FUNCTION is None:
+            is_auth = request.user.is_authenticated
+        else:
+            # There, most probably, is no user. If that's the case, we need to create him.
+            user_id = settings.HEADER_VALIDATION_FUNCTION(request.headers)
+            is_auth = user_id is not None
```

Comment:
So wait, this middleware will create a user in PG on first request from a client bearing that user's token?